client/client.go

```diff
@@ -448,3 +458,24 @@ func (c *CfClient) InterceptAddCluster(ctx context.Context, req *http.Request) e
 	req.URL.RawQuery = q.Encode()
 	return nil
 }
+
+type safeBool struct {
+	*sync.RWMutex
+	value bool
+}
+
+func newSafeBool(value bool) safeBool {
+	return safeBool{&sync.RWMutex{}, value}
+}
+
+func (s safeBool) set(value bool) {
+	s.RLock()
+	defer s.RUnlock()
+	s.value = value
+}
+
+func (s safeBool) bool() bool {
+	s.Lock()
+	defer s.Unlock()
+	return s.value
+}
```

Comment:
maybe you can use atomic package (int)? instead of custom implementation

Reply:
Had a look at atomic and will make a change to use it 👍